compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -1264,7 +1264,7 @@ impl<'hir> LoweringContext<'_, 'hir> {
             (Some(..), Some(..), HalfOpen) => hir::LangItem::Range,
             (None, Some(..), Closed) => hir::LangItem::RangeToInclusive,
             (Some(..), Some(..), Closed) => unreachable!(),
-            (_, None, Closed) => self.diagnostic().span_fatal(span, "inclusive range with no end"),
+            (_, None, Closed) => self.tcx.sess.emit_fatal(InclusiveRangeWithNoEnd { span }),
```

Comment:
Something like this:

```suggestion
            (start, None, Closed) => {
                self.tcx.sess.emit_err(InclusiveRangeWithNoEnd { span });
                match start {
                    Some(..) => hir::LangItem::RangeFrom,
                    None => hir::LangItem::RangeFull,
                }
            }
```